### PR TITLE
feat: allow seeding of the accounts in a permission request

### PR DESCRIPTION
**Pull request by FrederikBolding**
## **Description**

Allow seeding the permission request for a dapp connection with previously approved accounts using the following RPC request:
```
{
    "jsonrpc": "2.0",
    "method": "wallet_requestPermissions",
    "params": [
        {
            "eth_accounts": {
                "caveats": [
                    {
                        "type": "restrictReturnedAccounts",
                        "value": [
                            "0xc6d5a3c98ec9073b54fa0969957bd582e8d874bf"
                        ]
                    }
                ]
            }
        }
    ],
    "id": 0
}
```

If no caveat is found, we fall back to the previous logic of preselecting the currently selected account.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/24082?quickstart=1)

### CODE REVIEW

File: ui/pages/permissions-connect/permissions-connect.component.js [hunk 1]
```diff
@@ -30,6 +34,20 @@ import SnapResult from './snaps/snap-result';
...
+function getDefaultSelectedAccounts(currentAddress, permissionsRequest) {
+  const permission =
+    permissionsRequest.permissions[RestrictedMethods.eth_accounts];
+  const accountsCaveat = permission?.caveats?.find(
+    (caveat) => caveat.type === CaveatTypes.restrictReturnedAccounts,
+  )?.value;
+
+  if (accountsCaveat) {
+    return new Set(accountsCaveat.map((address) => address.toLowerCase()));
+  }
```
Comment (rekmarks) [inline ui/pages/permissions-connect/permissions-connect.component.js:46]:
```suggestion
  const requestedAccounts = permission?.caveats?.find(
    (caveat) => caveat.type === CaveatTypes.restrictReturnedAccounts,
  )?.value;

  if (requestedAccounts) {
    return new Set(requestedAccounts.map((address) => address.toLowerCase()));
  }
```

File: app/scripts/controllers/permissions/specifications.test.js [hunk 2]
```diff
@@ -231,31 +231,6 @@ describe('PermissionController specifications', () => {
...
-                    value: ['0x1', '0x2'],
-                  },
-                ],
-                invoker: 'foo.bar',
-                target: 'eth_accounts',
-              },
-              { approvedAccounts: ['0x1'] },
-            ),
-          ).toThrow(/Received unexpected caveats./u);
-        });
```
Comment (rekmarks) [inline app/scripts/controllers/permissions/specifications.test.js:258]: This should be replaced with a test that verifies the new behavior. Specifically, we should verify that the caveat on the permission request is ignored and only `approvedAccounts` is honored.

**Review by rekmarks — APPROVED** [review]
LGTM!

**Review by adonesky1 — APPROVED** [review]
LGTM

**Pull request merged**